gap/oper.gi

```diff
@@ -1333,6 +1333,55 @@ function(D, u, v)
   return out;
 end);
 
+InstallMethod(IsDigraphPath, "for a digraph and list",
+[IsDigraph, IsList],
+function(D, digraph_path_output)
+  if Length(digraph_path_output) <> 2 then
+    DError("the 2nd argument (a list) must have length 2, ",
+           "but found length {}",
+           Length(digraph_path_output));
+  fi;
+  return IsDigraphPath(D, digraph_path_output[1], digraph_path_output[2]);
+end);
+
+InstallMethod(IsDigraphPath, "for a digraph, homog. list, and homog. list",
+[IsDigraph, IsHomogeneousList, IsHomogeneousList],
+function(D, nodes, edges)
+  local a, err, i;
+  if Length(nodes) <> Length(edges) + 1 then
+    DError("the 2nd and 3rd arguments (lists) are incompatible, ",
+           "expected 3rd argument of length {}, got {}",
+           Length(nodes) - 1,
+           Length(edges));
+  fi;
+  a := OutNeighbours(D);
+  for i in [1 .. Length(edges)] do
+    if nodes[i] > Length(a) then
+      DError("the 2nd argument (a list) must consist of vertices ",
+             "of the 1st argument (a digraph), expected a value in ",
+             "the range [1, {}] but found {} in position {}",
+             Length(a),
+             nodes[i],
+             i);
+    elif edges[i] > Length(a[nodes[i]]) then
+      if not IsEmpty(a[nodes[i]]) then
+        err := "the range [1, {3}]";
+      else
+        err := "the range []";
+      fi;
+      DError("the 3rd argument (a list) must consist of values in ",
+             err,
+             " but found {1} in position {2}",
+             edges[i],
+             i,
+             Length(a[nodes[i]]));
+    elif a[nodes[i]][edges[i]] <> nodes[i + 1] then
+      return false;
```

Comment:
I disagree with the philosophy of your implementation (of course it's just my opinion so feel free to disagree).

In my opinion the error should only be given here for a 'type' error: so at the start of the method, you should check that the lists have compatible lengths and both consist of positive integers, and give an error if not.

Then, I think that if one of the vertices isn't in the digraph, or if the indexed out-neighbour doesn't actually exist, the result should be `false` rather than an error. At this point in the method, the input is at least a path in _some_ digraph (maybe a very big and weird one, but still), and the question is: is it a path in _this_ digraph: yes or no?

For me this fits in with other ways that GAP works, e.g.:
```
gap> T5 := FullTransformationMonoid(5);;
gap> P := PartitionMonoid(5);;
gap> IsSubsemigroup(T5, P);
false
```
(`false` rather than `Error...`).

A bit artificial, but a possible use case could be filtering a list of a paths of a big digraph `D1` by whether or not they also exist in a sub digraph `D2` of `D1`, something like `Filtered(paths, x -> IsDigraphPath(D2, x))` which probably wouldn't work in your current implementation.

```suggestion
    if nodes[i] > Length(a) or edges[i] > Length(a[nodes[i]]) or a[nodes[i]][edges[i]] <> nodes[i + 1] then
      return false;
```

Reply:
This makes sense, I'm happy to make this change!